### PR TITLE
fix for #58

### PR DESCRIPTION
**Pull request by dutronlabs**
I came across the issue recently for #58 as well, and saw the change was relatively simple. 

I added some tests in there as well to check the custom error handler was working okay, looks good on my end.

### CODE REVIEW

**Comment by jacebrowning** [comment]
It looks like this will need a way to handle both Flask 0.10.x and 0.11.x.

**Comment by dutronlabs** [comment]
Sounds good, I'll make those changes shortly.

File: flask_api/app.py [hunk 1]
```diff
@@ -116,3 +123,7 @@ def create_url_adapter(self, request):
                 self.config['SERVER_NAME'],
                 script_name=self.config['APPLICATION_ROOT'] or '/',
                 url_scheme=self.config['PREFERRED_URL_SCHEME'])
+
+    @staticmethod
+    def get_minor_version():
```
Comment (jacebrowning) [inline flask_api/app.py]: I wonder if this function might make more sense to live in `compat.py` and be reworked to something like `is_flask_0_10`?

File: flask_api/app.py [hunk 2]
```diff
@@ -88,9 +89,15 @@ def handle_user_exception(self, e):
         if handlers is not None:
             blueprint_handlers = handlers.get(None, ())
         app_handlers = self.error_handler_spec[None].get(None, ())
-        for typecheck, handler in chain(blueprint_handlers, app_handlers):
-            if isinstance(e, typecheck):
-                return handler(e)
+        flask_version.split()
```
Comment (jacebrowning) [inline flask_api/app.py]: I don't think this line has any affect.

File: flask_api/app.py (hunk 1 above)
Comment (dutronlabs) [inline flask_api/app.py]: sure, can do that.

File: flask_api/app.py (hunk 2 above)
Comment (dutronlabs) [inline flask_api/app.py]: will amend.

File: flask_api/app.py (hunk 1 above)
Comment (jacebrowning) [inline flask_api/app.py]: Or `is_flask_10` or `is_flask_legacy`. Whatever makes most sense to you.

**Comment by dutronlabs** [comment]
i see the issue, WIP

**Comment by jacebrowning** [comment]
👍 Looks good to me. All set to merge?

**Comment by dutronlabs** [comment]
Looks good to me.

**Pull request merged**